src/setuptools_pyproject_migration/__init__.py

```diff
@@ -71,8 +82,17 @@ def _generate_entry_points(entry_points: Optional[Dict[str, List[str]]]) -> Dict
 
     parsed_entry_points: Dict[str, Dict[str, str]] = {}
 
-    for eptype, raweps in entry_points.items():
-        parsed_entry_points[eptype] = dict(map(_parse_entry_point, raweps))
+    if isinstance(entry_points, str):
+        # INI-style
+        parser = configparser.ConfigParser()
+        parser.read_string(entry_points)
```

Comment:
Oh and I just noticed in the documentation that [`pkg_resources.EntryPoint.parse_map()`](https://setuptools.pypa.io/en/latest/pkg_resources.html#creating-and-parsing) is a thing. I'm guessing that's the same logic setuptools uses to parse these string entry point specifications, and if so we could use it. On the other hand it might be a good idea to avoid any dependencies on pkg_resources because it _is_ deprecated :shrug: but given that the whole point of this package is to work with deprecated-ish configuration files, maybe that's okay.

Reply:
Yeah, I did see mention of a `split_sections()` "utility" function, but so far haven't uncovered it.  Otherwise that's exactly what I would have used (or at least seen what its implementation was).

I just went digging for it now…
https://github.com/pypa/setuptools/blob/main/pkg_resources/__init__.py#L3361-L3384

Looks like it'd return `(None, [lines_of_text])` in the case of a missing section name, not `DEFAULT` like `configparser` does.

When I look at how it's used in `pkg_resources`, it looks like this is not a supported configuration:

https://github.com/pypa/setuptools/blob/main/pkg_resources/__init__.py#L2686-L2700

While we've got logic to parse out the individual lines that in theory duplicates what's being done here, it's code that's in the standard Python library.

In fact, one approach would be to use `configparser` to re-implement `_parse_entry_point`: use `"\n".join(raweps)` to bundle everything up and let `configparser` extract out the item values.

Reply:
When I tested, both `EntryPoint.parse_map()` and `ConfigParser.read_string()` raise a `ValueError` if the section name is completely missing, so I'm fine with not really worrying about how to handle that case. I mean, presumably people are running this on valid setuptools configurations which have been used to build and distribute real projects, so it kind of doesn't matter how we handle invalid configurations (as long as it's not, like, deleting the user's home directory :laughing:). At this point I'm thinking more about how to handle an explicitly written section which is actually called `DEFAULT`.

Reimplementing the whole thing with `ConfigParser` is interesting to consider, but personally I wouldn't... this is definitely an opinion, but I'd rather avoid "unparsing" and then parsing an object unless there's a really significant benefit we get from it, and I don't think this meets that threshold.

Reply:
Fair point … I can roll back that commit. :-)

Reply:
Actually, I note the `DEFAULT` section has special behaviour… when included, it populates all other sections with its content.

So if I do:

```
[DEFAULT]
ep0=mod:fn0

[type1]
ep1=mod:fn1
ep2=mod:fn2

[type2]
ep3=mod:fn3
ep4=mod:fn4
```

That gets passed through as:

```
[DEFAULT]
ep0=mod:fn0

[type1]
ep1=mod:fn1
ep2=mod:fn2
ep0=mod:fn0

[type2]
ep3=mod:fn3
ep4=mod:fn4
ep0=mod:fn0
```

To leave things as they are and treat `DEFAULT` like any other section, one must pass `default_section=None` in the constructor of `ConfigParser`.  Otherwise you get this "special" treatment.

https://stackoverflow.com/a/71891193

So if I do this, I can have a section named `DEFAULT`; and furthermore, if I strip "empty" groups, the result will behave the way we expect.

Reply:
`default_section=""` works, and keeps `mypy` happy.